PyRoute/DeltaReduce.py

```diff
@@ -0,0 +1,325 @@
+"""
+Created on May 23, 2023
+
+The core reduction engine used by delta debugging.
+Modify this class to add different reduction passes.
+
+@author: CyberiaResurrection
+"""
+import functools
+import logging
+import math
+
+from PyRoute.DeltaDictionary import DeltaDictionary
+from PyRoute.DeltaGalaxy import DeltaGalaxy
+from PyRoute.HexMap import HexMap
+from PyRoute.SpeculativeTrade import SpeculativeTrade
+from PyRoute.StatCalculation import StatCalculation
+from PyRoute.SubsectorMap2 import GraphicSubsectorMap
+
+
+class DeltaReduce:
+
+    def __init__(self, sectors, args, interesting_line=None, interesting_type=None):
+        assert isinstance(sectors, DeltaDictionary), "Sectors object must be an instance of DeltaDictionary"
+        self.sectors = sectors
+        self.args = args
+        # Interesting_line allows the caller to tighten the definition of interesting by requiring a specific
+        # string to appear in the exception message
+        # Interesting_type requires an exception type to contain a specific string
+        # If both are defined, they both have to match for the result to be interesting
+        self.interesting_line = interesting_line
+        self.interesting_type = interesting_type
+        self.logger = logging.getLogger('PyRoute.Star')
+        logging.disable(logging.WARNING)
+
+    def is_initial_state_interesting(self):
```

Comment:
Confirm that the original input is interesting (for whatever value the user specified) _before_ spending any time reducing it.  If the original input is non interesting, skip the hiding to nowhere.